apollo-router/src/apollo_studio_interop/mod.rs

```diff
@@ -473,10 +482,17 @@ impl UsageGenerator<'_> {
     }
 
     fn add_enum_reference(&mut self, enum_name: String, enum_value: String) {
-        self.enums_by_name
-            .entry(enum_name)
-            .or_default()
-            .insert(enum_value.to_string());
+        // Not using entry API here due to performance impact
+        let enum_name_stats = match self.enums_by_name.get_mut(&enum_name) {
+            Some(existing_stats) => existing_stats,
+            None => {
+                self.enums_by_name
+                    .insert(enum_name.to_string(), HashSet::new());
+                self.enums_by_name.get_mut(&enum_name).unwrap()
```

Comment:
Could we change the `unwrap()` with an `expect()` please ? Could be done in a follow up PR